src/cljs/harja/tiedot/kanavat/urakka/liikenne.cljs

```diff
@@ -649,49 +654,53 @@
 
   MuokkaaAluksia
   (process-event [{alukset :alukset v :virheita?} {tapahtuma :valittu-liikennetapahtuma :as app}]
-      (if tapahtuma
-        (-> app
-            (assoc-in [:valittu-liikennetapahtuma ::lt/alukset] (kasittele-suunta-alukselle tapahtuma alukset))
-            (assoc-in [:valittu-liikennetapahtuma :grid-virheita?] v))
-        app))
+    (if tapahtuma
+      (-> app
+        (assoc-in [:valittu-liikennetapahtuma ::lt/alukset] (kasittele-suunta-alukselle tapahtuma alukset))
+        (assoc-in [:valittu-liikennetapahtuma :grid-virheita?] v))
+      app))
 
   VaihdaSuuntaa
   (process-event [{alus :alus suunta :suunta} app]
     (let [uusi-suunta (if (:ei-suuntaa @lt/suunnat-atom)
-                        (cond (= :ylos suunta) :alas
-                              (= :alas suunta) :ei-suuntaa
-                              (= :ei-suuntaa suunta) :ylos
-                              :else :ylos)
-                        (cond (= :ylos suunta) :alas
-                              (= :alas suunta) :ylos
-                              :else :ylos))
+                        (cond
+                          (= :ylos suunta) :alas
+                          (= :alas suunta) :ei-suuntaa
+                          (= :ei-suuntaa suunta) :ylos
+                          :else :ylos)
+                        (cond
+                          (= :ylos suunta) :alas
+                          (= :alas suunta) :ylos
+                          :else :ylos))
 
           uusi (assoc alus ::lt-alus/suunta uusi-suunta)]
       (update app :valittu-liikennetapahtuma
-              (fn [t]
-                (update t ::lt/alukset
-                        (fn [alukset]
-                          (map #(if (sama-alusrivi? uusi %) uusi %) alukset)))))))
+        (fn [t]
+          (update t ::lt/alukset
+            (fn [alukset]
+              (map #(if (sama-alusrivi? uusi %) uusi %) alukset)))))))
 
   AsetaSuunnat
   (process-event [{suunta :suunta} app]
     (-> app
-        (assoc-in [:valittu-liikennetapahtuma :valittu-suunta] suunta)
-        (update-in [:valittu-liikennetapahtuma ::lt/alukset]
-                   (fn [alukset]
-                     (map #(assoc % ::lt-alus/suunta suunta) alukset)))))
+      (assoc-in [:valittu-liikennetapahtuma :valittu-suunta] suunta)
+      (update-in [:valittu-liikennetapahtuma ::lt/alukset]
+        (fn [alukset]
+          (map #(assoc % ::lt-alus/suunta suunta) alukset)))))
 
   TallennaLiikennetapahtuma
   (process-event [{t :tapahtuma} {:keys [tallennus-kaynnissa?] :as app}]
     (if-not tallennus-kaynnissa?
-      (let [params (-> (tallennusparametrit t)
-                       (assoc :hakuparametrit (hakuparametrit app)))]
+      (let [params (->
+                     (tallennusparametrit t)
```

Comment:
Koodi on helpompi lukea, jos threading makrolle annetaan -> jälkeen mappi, jota se käsittelee. Eli nosta tämä edelliselle riville.

Reply:
Ja nytkö katson tarkemmin, niin tarvitaanko tässä threading makroa?
Voisko se mennä vain näin: (assoc (tallennusparametrit t) :hakuparametrit (hakuparametrit app))

Reply:
Näin on, hyvä nosto. Korjattu ja testattu 
https://github.com/finnishtransportagency/harja/pull/2967/commits/a164be23fd71959573174917c9295719ed30bb7b